packages/material-ui-system/src/styled.test.js

```diff
@@ -308,6 +308,30 @@ describe('styled', () => {
       });
     });
 
+    it('variants should not be skipped if overridesResolver is not defined', () => {
```

Comment:
What do you mean by "skipped" here. The test looks like it's not using `variant` in any way and is also passing without the change.

Reply:
Should you maybe use a theme that has styles for this specific variant?

Reply:
Yes, it is used on line 322 https://github.com/mui-org/material-ui/pull/27859/files#diff-57eedb98ff493d66eb3e22d9e9ef1b1540504af969aa62accbb30b3a2a780044R322 The same theme is used in all tests. It is defined in https://github.com/mui-org/material-ui/blob/5585e100feec585f40f42ba2db605a56d1594718/packages/material-ui-system/src/styled.test.js#L140

Previously the `variants` from the theme were not applied, if when creating the component, the `overridesResolver` option was not applied. That's what I mean by should not be skipped.

Reply:
> Previously the variants from the theme were not applied, if when creating the component, the overridesResolver option was not applied.

But that's not related to the change in implementation in this PR. I ran the test with `yarn test:karma` and in JSDOM locally on `next` and it is passing.

This is confusing long-term because usually tests are coupled with the change in implementation. But here the change in implementation and test are unrelated.

Maybe you need to assert on different CSS properties to showcase what behavior changed?

Reply:
I see your point. I've fixed the tests. I was adding tests for a non root slot, but then variants are never applied :)